<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Description</TITLE>
<META NAME="description" CONTENT="Description">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node3.html">
<LINK REL="previous" HREF="node1.html">
<LINK REL="up" HREF="manual.html">
<LINK REL="next" HREF="node3.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html111"
  HREF="node3.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html107"
  HREF="manual.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html101"
  HREF="node1.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html109"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html112"
  HREF="node3.html">Requirements and installation</A>
<B> Up:</B> <A NAME="tex2html108"
  HREF="manual.html">get_homologues manual</A>
<B> Previous:</B> <A NAME="tex2html102"
  HREF="node1.html">Contents</A>
 &nbsp; <B>  <A NAME="tex2html110"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H1><A NAME="SECTION00020000000000000000">
Description</A>
</H1>

<P>
This document describes the software <SPAN  CLASS="textit">get_homologues</SPAN> and provides a few examples on how to install and use it. 
<SPAN  CLASS="textit">get_homologues</SPAN> is mainly written in the Perl programming language and includes several algorithms
designed for three main tasks:
<DL COMPACT>
<DT>1</DT>
<DD>Clustering protein and nucleotide sequences in homologous (possibly orthologous) groups, on the grounds of sequence similarity.
</DD>
<DT>2</DT>
<DD>Identification of orthologous groups of intergenic regions, flanked by orthologous open reading frames (ORFs), 
conserved across related genomes.
</DD>
<DT>3</DT>
<DD>Definition of pan- and core-genomes by calculation of overlapping sets of proteins.
</DD>
</DL>

<P>
While the program was mainly developed for the study of bacterial genomes in GenBank format, 
it can also be applied to eukaryotic sets of sequences, although in this case the meaning of terms such as core- 
or pan-genome might change, and intergenic regions might be much larger.  

<P>
The twin <SPAN  CLASS="textit">manual_get_homologues-est.pdf</SPAN> file describes the specific options of <SPAN  CLASS="textit">get_homologues-est</SPAN>, 
designed and tested for the task of clustering transcripts and, more generally, DNA-sequences of strains of the same species. 

<P>
<BR><HR>
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
