<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Estimating core/pan-genome size by sampling genomes</TITLE>
<META NAME="description" CONTENT="Estimating core/pan-genome size by sampling genomes">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node26.html">
<LINK REL="previous" HREF="node24.html">
<LINK REL="up" HREF="node21.html">
<LINK REL="next" HREF="node26.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html414"
  HREF="node26.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html410"
  HREF="node21.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html404"
  HREF="node24.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html412"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html415"
  HREF="node26.html">Calculating Pfam enrichment of</A>
<B> Up:</B> <A NAME="tex2html411"
  HREF="node21.html">Performing genome composition analyses</A>
<B> Previous:</B> <A NAME="tex2html405"
  HREF="node24.html">Calculating cloud, shell and</A>
 &nbsp; <B>  <A NAME="tex2html413"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H3><A NAME="SECTION00058400000000000000"></A><A NAME="composition"></A>
<BR>
Estimating core/pan-genome size by sampling genomes
</H3>

<P>
The pioneer work of Tettelin and collaborators 
(PubMed=http://www.ncbi.nlm.nih.gov/sites/ppmc/articles/PMC1216834/16172379) unveiled
that bacterial genomes are dynamic containers that harbour essential genes and also 
accessory elements, which might be unique to each community. <SPAN  CLASS="textit">get_homologues.pl</SPAN> can be used
to perform such genome composition analyses. 
The rationale is to sample a set of genomes (present in the input folder) and keep adding genome
after genome keeping track of i) the novel genes added to the pool and ii) the genes that fall in 
pre-existing clusters.
This sampling experiment can be done with any of the included 3 algorithms (please see Table <A HREF="node11.html#tab:algs">2</A>), 
by invoking option <code>-c </code>. For instance, we could try <code>$ ./get_homologues.pl -d sample_buch_fasta -c</code>:
<BR><PRE>
[... same as first example ...]

# genome composition report (samples=10,permutations=24)
# genomic report parameters: MIN_PERSEQID_HOM=0 MIN_COVERAGE_HOM=20
# genome order:
# 0 Buch_aph_APS.faa
# 1 Buch_aph_Bp.faa
# 2 Buch_aph_Cc.faa
# 3 Buch_aphid_Sg.faa

## sample 0 (Buch_aph_APS.faa | 0,1,2,3,)
# adding Buch_aph_APS.faa: core=574 pan=574
[...]

# pan-genome (number of genes, can be plotted with plot_pancore_matrix.pl)
# file=sample_buch_fasta_homologues/pan_genome_algBDBH.tab
genomes	mean	stddev	|	samples
0	490	90	|	574	507	507	546	357	357	546	574	357	574	
1	572	28	|	598	585	585	587	521	521	562	592	575	598	
2	597	5	|	606	593	594	594	594	596	600	599	591	606	
3	608	5	|	615	602	600	608	605	605	611	613	605	615	

# core-genome (number of genes, can be plotted with plot_pancore_matrix.pl)
# file=sample_buch_fasta_homologues/core_genome_algBDBH.tab
genomes	mean	stddev	|	samples
0	490	90	|	574	507	507	546	357	357	546	574	357	574	
1	420	82	|	466	466	466	523	324	324	317	523	324	466	
2	327	36	|	319	319	434	315	310	315	313	318	309	319	
3	310	4	|	313	313	313	311	304	304	311	313	304	313	

[... same as first example ...]
</PRE>

<P>

<DIV ALIGN="CENTER"><A NAME="fig:core"></A><A NAME="fig:pan"></A><A NAME="558"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 14:</STRONG>
Core-genome (left) and pan-genome (right) estimates after ten random samples of 4 taxa. 
Fitted curves follow functions first proposed by Tettelin in 2005 (PubMed=16172379).
Residual standard errors are reported on the right margin as a measure of the goodness of fit.</CAPTION>
<TR><TD><TABLE  WIDTH="50%">
<TR><TD>
<DIV ALIGN="CENTER">

</DIV><IMG
  WIDTH="480" HEIGHT="480" ALIGN="BOTTOM" BORDER="0"
 SRC="./core_genome_algBDBH.png"
 ALT="Image core_genome_algBDBH">
</TD></TR>
</TABLE>
 
<TABLE  WIDTH="50%">
<TR><TD>
<DIV ALIGN="CENTER">

</DIV><IMG
  WIDTH="480" HEIGHT="480" ALIGN="BOTTOM" BORDER="0"
 SRC="./pan_genome_algBDBH.png"
 ALT="Image pan_genome_algBDBH">
</TD></TR>
</TABLE></TD></TR>
</TABLE>
</DIV>

<P>
As can be seen, the output now contains two data frames which summarize the 
genome composition analysis done by sampling, which are also stored as tab-separated files.
These text files can be used to plot the core- and pan-genome, with help from
the accompanying script <SPAN  CLASS="textit">plot_pancore_matrix.pl</SPAN>. A suitable command would be: 
<BR><code> ./plot_pancore_matrix.pl -i sample_buch_fasta_homologues/core_genome_algBDBH.tab -f core_Tettelin</code>
<BR>
<P>
The script also supports the core function as modified by Willenbrock and 
collaborators (PubMed=http://www.ncbi.nlm.nih.gov/pmc/articles/PMC224626918088402), 
as shown on the next figure in a more realistic set of 35 genomes. Both fits can be superimposed 
by calling option <code> -f core_both</code>.

<P>
Besides standard core- and pan-genomes, it is possible to estimate the evolution of the soft core-genome, 
which is a relaxed version of the core that considers genes found in a fraction (by default 0.95) of genomes, and thus accommodates 
some annotation or assembly errors. This experiment can be done with either the OMCl or COGS algorithms 
by invoking options <code>-c -z</code>. The resulting data file can be plotted the same way.

<P>

<DIV ALIGN="CENTER"><A NAME="fig:core2"></A><A NAME="566"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 15:</STRONG>
Core-genome estimate after ten random samples of 35 taxa.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="540" HEIGHT="540" ALIGN="BOTTOM" BORDER="0"
 SRC="./sample_core_genome2.png"
 ALT="Image sample_core_genome2">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>
The genome composition analyses presented so far are actually random sampling experiments. 
It is thus worth mentioning that the user can control the order in which genomes are sampled during
these simulations, by enforcing a list of genomes with option <code>-I</code>, already introduced in section <A HREF="node20.html#intergen">4.7</A>,
or by setting the seed of the random number generator with option <code>-R</code>.
In the first case only one sampling is performed and therefore the standard deviation of the core and pan values is zero.
The second strategy ensures that sampling order is conserved in different program executions and thus allows merging CDS 
core-genomes and non-coding genes (such as rRNAs) core-genomes computed separately over the same set of taxa, with help from
accompanying script <SPAN  CLASS="textit">add_pancore_matrices.pl</SPAN>.

<P>

<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A NAME="tex2html414"
  HREF="node26.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html410"
  HREF="node21.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html404"
  HREF="node24.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html412"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html415"
  HREF="node26.html">Calculating Pfam enrichment of</A>
<B> Up:</B> <A NAME="tex2html411"
  HREF="node21.html">Performing genome composition analyses</A>
<B> Previous:</B> <A NAME="tex2html405"
  HREF="node24.html">Calculating cloud, shell and</A>
 &nbsp; <B>  <A NAME="tex2html413"
  HREF="node1.html">Contents</A></B> </DIV>
<!--End of Navigation Panel-->
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
