<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Downstream analyses</TITLE>
<META NAME="description" CONTENT="Downstream analyses">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="previous" HREF="node32.html">
<LINK REL="up" HREF="node30.html">
<LINK REL="next" HREF="node34.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html507"
  HREF="node34.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html503"
  HREF="node30.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html499"
  HREF="node32.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html505"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html508"
  HREF="node34.html">Frequent warnings and error</A>
<B> Up:</B> <A NAME="tex2html504"
  HREF="node30.html">Frequently asked questions (FAQs)</A>
<B> Previous:</B> <A NAME="tex2html500"
  HREF="node32.html">Run options</A>
 &nbsp; <B>  <A NAME="tex2html506"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H2><A NAME="SECTION00063000000000000000">
Downstream analyses</A>
</H2>

<P>

<UL>
<LI>What are pancore and pangenome files/matrices?

<P>
Pancore matrices contain estimates of core- and pan-genomes and they are produced by <SPAN  CLASS="textit">get_homologues.pl</SPAN> with option <code>-c</code>.
These files take names such as <code>pan_genome_algBDBH_C75.tab</code>, which record the algorithm employed, and are generated by random-sampling genomes. 
Sampling can be controlled and reproduced by using the same random-number generator seed  (see section <code>-R</code> in Section <A HREF="node11.html#options">3.4</A>). 
Such files can be used to render plots (and fitted functions) with script <code>plot_pancore_matrix.pl</code>, as shown in Section <A HREF="node25.html#composition">4.8.4</A>.

<P>
Instead, pangenome matrices are generated by accessory script <SPAN  CLASS="textit">compare_clusters.pl</SPAN>, and contain information about what which genomes contain sequences
from gene clusters, with no sampling involved (see Section <A HREF="node22.html#pange">4.8.1</A>).
They take names such as <code>pangenome_matrix_t0.tab</code>.

<P>
</LI>
<LI>Is it possible to plot pangenome matrices with <SPAN  CLASS="textit">compare_clusters.pl</SPAN> using a single cluster directory?

<P>
Yes, no problem. The script will generate the <code>intersection_t0.cluster_list</code> , <code>pangenome_matrix_t0.phylip</code> and <code>pangenome_matrix_t0.tab</code> 
files based on the clusters found by the chosen algorithm. The only thing you won't find in the directory are Venn diagrams, since at least 2 cluster sets 
(generated by 2 algorithms) are required to compute them.

<P>
</LI>
<LI>What is the advantage of providing multiple cluster output directories to the -d 'dir1,dir2,dir3'  option of <SPAN  CLASS="textit">compare_clusters.pl</SPAN>?

<P>
When provided with the output directories holding the clusters generated by 2 or 3 algorithms, the script will select only those clusters that 
contain exactly the same sequences in each of the clusters. This may be valuable for example if the user is interested in defining a very robust 
core genome set, containing only those families with exactly the same members, independently of the chosen algorithm. However, it is possible that 
several otherwise important families get lost for downstream analyses, such as presence-absence analyses of gene families in pairs of lineages, 
which can be done with <SPAN  CLASS="textit">parse_pangenome_matrix.pl</SPAN>. 

<P>
</LI>
<LI>We have sequenced 25 bacteria genomes and used <SPAN  CLASS="textit">get_homologues.pl</SPAN> to get orthologs included in all the 25 strains. 
As I already used BDBH method for core genome analysis, now I cannot switch to COG method for pangenome matrix generation.
Could you indicate me how to get the full pangenome matrix using BDBH method?

<P>
BDBH results cannot be used for pangenome analysis, but you could re-run the software with the same 25 input genomes, now adding <code>-t 0 -M</code> for OMCL,
probably the best choice for such an analysis. This will re-use all blast results previously calculated and resume until OMCL analysis is completed.
Usually core sets produced by BDBH, COGS and OCML are very similar. Therefore, most of your previously tested core genes should also be picked up 
by OCML on this second run, and presumably you could now proceed to pangenome analysis.

<P>
</LI>
<LI>I am a PhD student doing some work on different strains. My input is .gbk from draft genomes and my aim at this point is to see what 
are core-genes in each of 3 groups and which of these are shared with the other 2 groups. I have been working under the impression that I 
can make 3 different groups and compare them or can I only make 1 big group and compare the results for different methods?

<P>
You have calculated 3 core-genomes from 3 different sets of strains and now you want to know the subset of core-genes present 
in all individual core-genomes. If you check compare_clusters.pl option you'll see that it says
"by default cluster dirs are expected to be derived from the same taxa", which is exactly what's failing in your examples.
If you want to find common core-clusters present if all 3 sets and also those shared by only some of the taxa I guess you should build a
pangenome matrix by running get_homologues.pl with all strains together with option -t 0. The matrix itself will give you the clusters 
present/absent in all and some of the strains, which I guess is what you need. The script parse_pangenome_matrix.pl can read this matrix 
and further help you identify these clusters.

<P>
</LI>
<LI>Could you please give a use case for <SPAN  CLASS="textit">compare_clusters.pl -r</SPAN>?

<P>
Option <code>-r</code> can be used to compare a list of core genes from a single genome G,
that is, with clusters containing only sequences from G, to clusters of a larger group of taxa (A,B,C,...,G) that includes G.

<P>
</LI>
<LI>The parsimony tree produced by <code>compare_cluster.pls -T</code> cannot be opened by MEGA.

<P>
Program PARS from the http://evolution.genetics.washington.edu/phylip/doc/pars.htmlPHYLIP suite often produces several 
alternative parsimony trees contained in the <code>pangenome_matrix_t0.phylip.ph</code> file, one per line. Some phylogeny programs, such as
http://www.megasoftware.netMEGA, require splitting these trees in separate files in order to properly read and plot them.

<P>
</LI>
<LI>Parsing a pangenome matrix with A &amp; B lists yields zero clusters.

<P>
When a command such as <code>./parse_pangenome_matrix.pl -m pangenome_matrix_t0.tab -A A.txt -B B.txt -g</code> is invoked, 
the passed A &amp; B files must contain taxon names matching exactly those of corresponding input files, including the extension. Instead of
a list such as:
<PRE>
taxonA 
taxonB_001
...
taxonZ244
</PRE>

<P>
the list should be:
<PRE>
taxonA.faa
taxonB_001.faa
...
taxonZ244.faa
</PRE>

<P>
</LI>
<LI>How does <SPAN  CLASS="textit">make_nr_pangenome_matrix.pl</SPAN> work?

<P>
Please refer to <SPAN  CLASS="textit">manual_get_homologues-est.pdf</SPAN> for an example application of this script.

<P>
</LI>
<LI>How does <SPAN  CLASS="textit">annotate_cluster.pl</SPAN> work?

<P>
Please refer to <SPAN  CLASS="textit">manual_get_homologues-est.pdf</SPAN> for an example application of this script.

<P>
</LI>
</UL>

<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A NAME="tex2html507"
  HREF="node34.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html503"
  HREF="node30.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html499"
  HREF="node32.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html505"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html508"
  HREF="node34.html">Frequent warnings and error</A>
<B> Up:</B> <A NAME="tex2html504"
  HREF="node30.html">Frequently asked questions (FAQs)</A>
<B> Previous:</B> <A NAME="tex2html500"
  HREF="node32.html">Run options</A>
 &nbsp; <B>  <A NAME="tex2html506"
  HREF="node1.html">Contents</A></B> </DIV>
<!--End of Navigation Panel-->
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
