<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Frequent warnings and error messages</TITLE>
<META NAME="description" CONTENT="Frequent warnings and error messages">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node35.html">
<LINK REL="previous" HREF="node30.html">
<LINK REL="up" HREF="manual.html">
<LINK REL="next" HREF="node35.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html519"
  HREF="node35.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html515"
  HREF="manual.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html509"
  HREF="node33.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html517"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html520"
  HREF="node35.html">Credits and references</A>
<B> Up:</B> <A NAME="tex2html516"
  HREF="manual.html">get_homologues manual</A>
<B> Previous:</B> <A NAME="tex2html510"
  HREF="node33.html">Downstream analyses</A>
 &nbsp; <B>  <A NAME="tex2html518"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H1><A NAME="SECTION00070000000000000000">
Frequent warnings and error messages</A>
</H1> 

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="757"></A>
<TABLE>
<CAPTION><STRONG>Table 4:</STRONG>
Frequent warnings and error messages produced by get_homologues and kin scripts.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<TABLE CELLPADDING=3 BORDER="1" ALIGN="CENTER">
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>error message</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>practical meaning</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>EXIT : cannot find previous input file XXXX, please re-run everything</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>This can happen when re-running the program with an input <code>-d directory</code>
which used to contain more sequences files, or with different names. This prevents the software
to recycle previous results, as it cannot ensure that sequences are still numbered consistently.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>WARNING: could not extract nucleotide sequences from file XXXX</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>You'll see this warning when using an uncomplete input GenBank file, lacking the nucleotide sequence at the bottom.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>WARNING: can only extract genes (not CDSs) from file XXXX</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>Occurs when reading a GenBank file lacking CDS features.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>WARNING: cannot use nucleotide sequences in file XXXX as they do not match those in file YYYY</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>This warning occurs when a twin XXXX <code>.fna</code> file (see Table <A HREF="node11.html#tab:informats">1</A>) contains a different number of sequences
than the corresponding YYYY <code>.faa</code> file, and cannot therefore be safely used to compile DNA clusters.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>EXIT, XXXX does not exist, Pfam search failed...</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>Occurs when a Pfam job submitted to the cluster (option <code>-D</code>) failed to report back and terminate. 
The solution is often to re-run the program, as it will only re-submit the missing Pfam jobs.
When solving problems with submitting jobs to the cluster queue it is helpful to check the .queue files.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>EXIT: cannot format BLAST sequence base</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>Happens when for some reason the collection of input sequences could not be formatted for BLAST. 
This might surface hard drive trouble or simply an architecture issue.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>EXIT, XXXX.blastout does not exist, BLAST search failed...</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>Again a BLAST error, spotted for failing to produce a BLAST output. Often the solution is simply to re-run,
as this might be simply a cluster overload problem. 
When solving problems with submitting jobs to the cluster queue it is often helpful top check the .queue files.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>EXIT: parsed XXXX output (YYYY) seems to be empty, please remove 'input_homologues/' and re-run</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>Another BLAST/Pfam error, which can happen if the programs fails to parse the results.
The simplest solution is usually to do as suggested and re-run.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>WARNING: please remove/rename results directory: XXXX if you change the sequences in your .gbk/.faa files or want to re-run</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>This warning is issued only to make it clear that the program is recycling previous BLAST results, 
which is usually a good idea, unless you specifically changed the contents of your input files 
(which should't be that common).</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>EXIT: cannot compile intergenic clusters as not all input GenBank files are valid</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>This message appears when the user requested intergenic clusters (option <code>-g </code>) but not all
parsed GenBank files contained nucleotide sequences. The solution is to check the input files and 
correct the offending one, which likely is uncomplete and lacks the nucleotide sequence at the bottom.</TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=180>WARNING: skipping cluster 123_XXX.fna , seems to duplicate 456_YYY.fna</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=252>This is issued by <SPAN  CLASS="textit">compare_clusters.pl</SPAN> when it finds, usually singleton, clusters with identical sequences produced by
the COG or OMCL algorithms. This can happen when such clusters contain short sequences, or perhaps with composition biases, that
yield few or even no BLAST hits when compared to all other sequences in a given setup. As these kinds of clusters can confound 
posterior analysis they are currently ignored by <SPAN  CLASS="textit">compare_clusters.pl</SPAN>.</TD>
</TR>
</TABLE>
</DIV>

<A NAME="tab:warns"></A></TD></TR>
</TABLE>
</DIV><P></P>
<BR>

<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A NAME="tex2html519"
  HREF="node35.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html515"
  HREF="manual.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html509"
  HREF="node33.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html517"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html520"
  HREF="node35.html">Credits and references</A>
<B> Up:</B> <A NAME="tex2html516"
  HREF="manual.html">get_homologues manual</A>
<B> Previous:</B> <A NAME="tex2html510"
  HREF="node33.html">Downstream analyses</A>
 &nbsp; <B>  <A NAME="tex2html518"
  HREF="node1.html">Contents</A></B> </DIV>
<!--End of Navigation Panel-->
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
