<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Program options</TITLE>
<META NAME="description" CONTENT="Program options">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node12.html">
<LINK REL="previous" HREF="node10.html">
<LINK REL="up" HREF="node7.html">
<LINK REL="next" HREF="node12.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html225"
  HREF="node12.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html221"
  HREF="node7.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html215"
  HREF="node10.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html223"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html226"
  HREF="node12.html">Accompanying scripts</A>
<B> Up:</B> <A NAME="tex2html222"
  HREF="node7.html">User manual</A>
<B> Previous:</B> <A NAME="tex2html216"
  HREF="node10.html">(Eukaryotic) FASTA amino acid</A>
 &nbsp; <B>  <A NAME="tex2html224"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H2><A NAME="SECTION00044000000000000000"></A> <A NAME="options"></A>
<BR>
Program options
</H2>

<P>
<PRE>
-v print version, credits and checks installation
-d directory with input FASTA files ( .faa / .fna ),           (overrides -i,
   GenBank files ( .gbk ), 1 per genome, or a subdirectory      use of pre-clustered sequences
   ( subdir.clusters / subdir_ ) with pre-clustered sequences   ignores -c, -g)
   ( .faa / .fna ); allows for new files to be added later;    
   creates output folder named 'directory_homologues'
-i input amino acid FASTA file with [taxon names] in headers,  (required unless -d is set)
   creates output folder named 'file_homologues'
   
   
Optional parameters:
-o only run BLAST/Pfam searches and exit                       (useful to pre-compute searches)
-c report genome composition analysis                          (follows order in -I file if enforced,
                                                                ignores -r,-t,-e)
-R set random seed for genome composition analysis             (optional, requires -c, example -R 1234,
                                                                required for mixing -c with -c -a runs)
-s save memory by using BerkeleyDB; default parsing stores
   sequence hits in RAM
-m runmode [local|cluster]                                     (default local)
-n nb of threads for BLAST/HMMER/MCL in 'local' runmode        (default=2)
-I file with .faa/.gbk files in -d to be included              (takes all by default, requires -d)


Algorithms instead of default bidirectional best-hits (BDBH):
-G use COGtriangle algorithm (COGS, PubMed=20439257)           (requires 3+ genomes|taxa)
-M use orthoMCL algorithm (OMCL, PubMed=12952885)


Options that control sequence similarity searches:
-C min %coverage in BLAST pairwise alignments                  (range [1-100],default=75)
-E max E-value                                                 (default=1e-05,max=0.01)
-D require equal Pfam domain composition                       (recommended with -m cluster
   when defining similarity-based orthology
-S min %sequence identity in BLAST query/subj pairs            (range [1-100],default=1 [BDBH|OMCL])
-N min BLAST neighborhood correlation PubMed=18475320          (range [0,1],default=0 [BDBH|OMCL])
-b compile core-genome with minimum BLAST searches             (ignores -c [BDBH])


Options that control clustering:
-t report sequence clusters including at least t taxa          (default t=numberOfTaxa,
                                                                t=0 reports all clusters [OMCL|COGS])
-a report clusters of sequence features in GenBank files       (requires -d and .gbk files,
   instead of default 'CDS' GenBank features                    example -a 'tRNA,rRNA',
                                                                NOTE: uses blastn instead of blastp,
                                                                ignores -g,-D)
-g report clusters of intergenic sequences flanked by ORFs     (requires -d and .gbk files)
   in addition to default 'CDS' clusters
-f filter by %length difference within clusters                (range [1-100], by default sequence
                                                                length is not checked)
-r reference proteome .faa/.gbk file                           (by default takes file with
                                                                least sequences; with BDBH sets
                                                                first taxa to start adding genes)
-e exclude clusters with inparalogues                          (by default inparalogues are
                                                                included)
-x allow sequences in multiple COG clusters                    (by default sequences are allocated
                                                                to single clusters [COGS])
-F orthoMCL inflation value                                    (range [1-5], default=1.5 [OMCL])
-A calculate average identity of clustered sequences,          (optional, creates tab-separated matrix,
 by default uses blastp results but can use blastn with -a      recommended with -t 0 [OMCL|COGS])
-z add soft-core to genome composition analysis                (optional, requires -c [OMCL|COGS])
</PRE>

<P>

<DIV ALIGN="CENTER"><A NAME="fig:flow"></A><A NAME="216"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 2:</STRONG>
Flowchart of get_homologues.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="3667" HEIGHT="5820" ALIGN="BOTTOM" BORDER="0"
 SRC="./flow.png"
 ALT="Image flow">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>
Typing <code>$ ./get_homologues.pl -h</code> on the terminal will show the available options, shown on the previous pages.

<P>
The only required option is either <code>-i</code>, used to choose an input file, or <code>-d</code> instead, which indicates 
an input folder, as seen in section <A HREF="node8.html#input">3.1</A>. It is important to remark that in principle only files 
with extensions <code>.faa</code> and <code>.gbk</code> are considered when parsing the <code>-d</code> directory. 
By using <code>.faa</code> input files in theory you might only calculate clusters of protein sequences.
In contrast, the advantage of using <code>.gbk</code> files is that you obtain both nucleotide and protein clusters.
If both types of input files are combined, only protein clusters will be produced.
However, if each input <code>.faa</code> file has a twin <code>.fna</code> file in place, containing the corresponding
nucleotide sequences in the same order, the program will attempt to produce the corresponding clusters of nucleotide sequences.
The possible input file combinations are summarized in Table <A HREF="#tab:informats">1</A>:

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="228"></A>
<TABLE>
<CAPTION><STRONG>Table 1:</STRONG>
Valid input file combinations.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<TABLE CELLPADDING=3 BORDER="1" ALIGN="CENTER">
<TR><TD ALIGN="LEFT">input file extensions</TD>
<TD ALIGN="CENTER">output clusters</TD>
</TR>
<TR><TD ALIGN="LEFT"><code>.gbk</code></TD>
<TD ALIGN="CENTER">amino acid + DNA sequence</TD>
</TR>
<TR><TD ALIGN="LEFT"><code>.faa</code></TD>
<TD ALIGN="CENTER">amino acid sequence</TD>
</TR>
<TR><TD ALIGN="LEFT"><code>.gbk</code> &amp; <code>.faa</code></TD>
<TD ALIGN="CENTER">amino acid sequence</TD>
</TR>
<TR><TD ALIGN="LEFT"><code>.faa</code> &amp; <code>.fna</code></TD>
<TD ALIGN="CENTER">amino acid + DNA sequence</TD>
</TR>
<TR><TD ALIGN="LEFT"><code>.gbk</code> &amp; <code>.faa</code> &amp; <code>.fna</code></TD>
<TD ALIGN="CENTER">amino acid + DNA sequence</TD>
</TR>
</TABLE>
</DIV>

<A NAME="tab:informats"></A></TD></TR>
</TABLE>
</DIV><P></P>
<BR>

<P>
The use of an input folder or directory (<code>-d</code>)
is recommended as it allows for new files to be added there in the future, reducing the computing required
for updated analyses. For instance, if a user does a first analysis with 5 input genomes today, it is possible
to check how the resulting clusters would change when adding an extra 10 genomes tomorrow, by copying these new 10 
<code>.faa</code> / <code>.gbk</code> input files to the pre-existing <code>-d</code> folder, so that all previous BLAST searches are re-used.

<P>
In addition to <code>.gbk</code> and <code>.faa</code> files, the input directory can also contain <SPAN  CLASS="textbf">one subfolder</SPAN> with pre-clustered sequences.
This feature was designed so that users can add previously produced <SPAN  CLASS="textit">get_homologues</SPAN> clusters, 
or any other set of grouped sequences in FASTA format, to be analysed. For such a subfolder to be recognized, it must
be named <code>subdir.clusters</code> or <code>subdir_</code>. Sample data folder <code>sample_buch_fasta/</code> contains such an example subfolder
which can be uncompressed to be tested. It is important to note that, during subsequent calculations, 
<SPAN  CLASS="textbf">these clusters are represented by the first sequence found in each</SPAN>. However, the output of the program will include 
all pre-clustered sequences for convenience. 

<P>
All remaining flags are options that can modify the default behavior of the program, which is to use the 
bidirectional best hit algorithm (BDBH) in order to compile clusters of potential orthologous ORFs,
taking the smallest genome as a reference. By default protein sequences are used to guide the clustering, thus
relying on BLASTP searches.

<P>
Perhaps the most important optional parameter would be the choice of clustering algorithm (Table <A HREF="#tab:algs">2</A>):
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="246"></A>
<TABLE>
<CAPTION><STRONG>Table 2:</STRONG>
List of available clustering algorithms.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<TABLE CELLPADDING=3 BORDER="1" ALIGN="CENTER">
<TR><TD ALIGN="LEFT">name</TD>
<TD ALIGN="LEFT">option</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=360>&nbsp;</TD>
</TR>
<TR><TD ALIGN="LEFT">BDBH</TD>
<TD ALIGN="LEFT">default</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=360>Starting from a reference genome, keep adding genomes stepwise while storing the sequence clusters
that result of merging the latest bidirectional best hits, as illustrated in Figure <A HREF="#BDBHflow">3</A>.</TD>
</TR>
<TR><TD ALIGN="LEFT">COGS</TD>
<TD ALIGN="LEFT"><code>-G</code></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=360>Merges triangles of inter-genomic symmetrical best matches, as described in 
PubMed=http://www.ncbi.nlm.nih.gov/sites/ppmc/articles/PMC2881409/20439257. 
Note that a single sequence might occasionally be included in several COGS clusters with option <code>-x</code>.</TD>
</TR>
<TR><TD ALIGN="LEFT">OMCL</TD>
<TD ALIGN="LEFT"><code>-M</code></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=360>OrthoMCL v1.4, uses the Markov Cluster Algorithm to group sequences,
with inflation (<code>-F</code>) controlling cluster granularity, as described in 
PubMed=http://www.ncbi.nlm.nih.gov/sites/ppmc/articles/PMC403725/12952885.</TD>
</TR>
</TABLE>
</DIV>

<A NAME="tab:algs"></A></TD></TR>
</TABLE>
</DIV><P></P>
<BR>

<P>

<DIV ALIGN="CENTER"><A NAME="BDBHflow"></A><A NAME="252"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 3:</STRONG>
Flowchart of BDBH algorithm with default parameters and G genomes. 
First, inparalogues, defined as intra-specific bidirectional best hits (BDBHs), are identified in each genome;
second, new genomes are incrementally compared to the reference genome and their BDBHs annotated;
finally, clusters containing at least 1 sequence per genome are conserved.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="1988" HEIGHT="3087" ALIGN="BOTTOM" BORDER="0"
 SRC="./BDBH.png"
 ALT="Image BDBH">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>
The remaining options are now reviewed:

<P>

<UL>
<LI>Apart from showing the credits, option <code>-v</code> can be helpful after installation, 
for it prints the enabled features of the program, which of course depend on the required and optional 
binaries mentioned in sections <A HREF="node5.html#binaries">2.2</A> and <A HREF="node6.html#dependencies">2.3</A>.

<P>
</LI>
<LI><code>-o</code> is ideally used to submit to a computer cluster the required BLAST (and Pfam) searches, preparing a job for posterior 
analysis on a single computer.

<P>
</LI>
<LI><code>-c</code> is used to request a pan- and core-genome analysis of the input genomes, which will be output as a tab-separated data file. 
The number of samples for the genome composition analysis is set to 10 by default, but this can be edited at the header of 
<code>get_homologues.pl</code> (check the <code>$NOFSAMPLESREPORT</code> variable). In addition, variables <code>$MIN_PERSEQID_HOM</code> and 
<code>$MIN_COVERAGE_HOM</code>, with default values 0 and 20, respectively, control how homologues are called. These values can
as well be edited to require more stringent identity and coverage requirements in order to call a sequence homologous, and therefore, redundant.

<P>
</LI>
<LI><code>-R</code> takes a number that will be used to seed the random generator used with option <code>-c</code>. By using the
same seed in different <code>-c</code> runs the user ensures that genomes are sampled in the same order.

<P>
</LI>
<LI><code>-s</code> can be used to reduce the memory footprint, provided that the Perl module 
http://search.cpan.org/perldoc?BerkeleyDBBerkeleyDB is in place (please check section <A HREF="node6.html#dependencies">2.3</A>). 
This option usually makes <SPAN  CLASS="textit">get_homologues</SPAN> slower, but for very large datasets or in machines with little memory resources 
this might be the only way to complete a job.

<P>
</LI>
<LI><code>-m</code> allows the choice of runmode, which can be either <code>-m local</code> (the default) or 
<code>-m cluster</code>. In the second case global variable <code>$SGEPATH</code> might need to be appropriately set, as explained in section <A HREF="node6.html#dependencies">2.3</A>,
as well as <code>$QUEUESETTINGS</code>, that specificies for instance a particular queue name for your cluster jobs.

<P>
</LI>
<LI><code>-n</code> sets the number of threads/CPUs to dedicate to each BLAST/HMMER job run locally, which by default is 2.

<P>
</LI>
<LI><code>-I list_file.txt</code> allows the user to restrict a <SPAN  CLASS="textit">get_homologues</SPAN> job to a subset of the genomes included in the input <code>-d </code> folder. 
This flag can be used in conjunction with <code>-c</code> to control the order in which genomes are considered during pan- and core-genome analyses.
Taking the <code>sample_buch_fasta</code> folder, a valid <code>list_file.txt</code> could contain these lines:
<PRE>
Buch_aph_APS.faa  
Buch_aph_Bp.faa  
Buch_aph_Cc.faa
</PRE>

<P>
</LI>
<LI>option <code>-C</code> sets the minimum percentage of coverage required to call two sequences best hits, 
as illustrated in the figure. The larger these values get, the smaller the chance that two sequences are 
found to be reciprocal  best hits. The default coverage value is set to 75%. 
This parameter has a large impact on the results obtained and its optimal values will depend on the input 
data and the anticipated use of the produced clusters:

<P>

<DIV ALIGN="CENTER"><A NAME="270"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 4:</STRONG>
Coverage [BDBH,OMCL] and overall segment coverage [COGS] illustrated with the
alignment of sequence 'query' to two aligned fragments of sequence 'subject', where 1,s1,e1,s2,e2 and L 
are alignment coordinates.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="106" HEIGHT="58" ALIGN="BOTTOM" BORDER="0"
 SRC="./match_cover.png"
 ALT="Image match_cover">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>
</LI>
<LI><code>-E</code> sets the maximum expectation value (E-value) for BLAST alignments. This value is by default set to 1e-05. 
This parameter might be adjusted for nucleotide BLAST searches or for very short proteins, under 40 residues.

<P>
</LI>
<LI><code>-S</code> can be passed to require a minimum % sequence identity for two sequences to be called best hits. 
This option does not affect COGS runs; its default value is set to 1. 

<P>
</LI>
<LI><code>-N</code> sets a minimum neighborhood correlation, as defined in 
PubMed=http://www.ncbi.nlm.nih.gov/sites/ppmc/articles/PMC2377100/18475320,
for two sequences to be called best hits. In this context 'neighborhood' is the set of homologous sequences reported by BLAST,
with the idea that two reliable best hits should have similar sets of homologous sequences.

<P>
</LI>
<LI><code>-D</code> is an extra restriction for calling best hits, that should have identical Pfam domain compositions. Note that this
option requires scanning all input sequences for Pfam domains, and this task requires some software to be installed (see section <A HREF="node6.html#dependencies">2.3</A>)
and extra computing time, ideally on a computer cluster (<code>-m cluster</code>). 
While for BDBH domain filtering is done at the time bidirectional best hits are called, this processing step is performed only after the 
standard OMCL and COGS algorithms have completed, to preserve each algorithm features.

<P>
</LI>
<LI><code>-b</code> reduces the number of pairwise BLAST searches performed while compiling core-genomes with algorithm BDBH,
reducing considerably memory and run-time requirements (for <SPAN CLASS="MATH"><IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img1.png"
 ALT="$G$"></SPAN> genomes, 3G searches are launched instead of the default <SPAN CLASS="MATH"><IMG
 WIDTH="24" HEIGHT="16" ALIGN="BOTTOM" BORDER="0"
 SRC="img2.png"
 ALT="$G^{2}$"></SPAN>).
It comes at the cost of being less exhaustive in finding inparalogues, but in our bacterial benchmarks this potential, undesired 
effect was negligible.

<P>

<DIV ALIGN="CENTER"><A NAME="fig:BDBHmin"></A><A NAME="791"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 5:</STRONG>
For  <SPAN CLASS="MATH"><IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img1.png"
 ALT="$G$"></SPAN> genomes, a typical get_homologues job requires running <SPAN CLASS="MATH"><IMG
 WIDTH="24" HEIGHT="16" ALIGN="BOTTOM" BORDER="0"
 SRC="img2.png"
 ALT="$G^{2}$"></SPAN> BLAST searches in order to compare all against all sequences,
including against itself to help infer inparalogues. Therefore, the resources required for calculating BLAST jobs grow quadratically. 
Instead, the BDBH algorithm with option -b requires only 3G BLAST searches (in grey) for any reference genome.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="1670" HEIGHT="868" ALIGN="BOTTOM" BORDER="0"
 SRC="./BDBHmin.png"
 ALT="Image BDBHmin">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>
</LI>
<LI><code>-t</code> is used to control which sequence clusters should be reported. By default only clusters which include at least one sequence 
per genome are output. However, a value of <code>-t 2</code> would report all clusters containing sequences from at least 2 taxa. A especial
case is <code>-t 0</code>, which will report all clusters found, even those with sequences from a single genome. 

<P>
</LI>
<LI><code>-a</code> forces the program to extract user-selected sequence features typically contained in GenBank files, such as 
tRNA or rRNA genes, instead of default CDSs. When using this option clusters are compiled by comparing nucleotide sequences
with BLASTN. Note that such BLASTN searches are expected to be less sensitive than default BLASTP searches.

<P>
</LI>
<LI><code>-g</code> can be used to request the compilation of clusters of intergenic sequences. This implies the calculation of ORF clusters and then
a search for pairs of 'orthologous' ORFs which flanking conserved intergenic regions, with the constraints set by three global variables in the
header of <code>get_homologues.pl</code>: <A NAME="intergdefs"></A><PRE>
my $MININTERGENESIZE     = 200;   # minimum length (nts) required for intergenic 
                                  # segments to be considered
my $MAXINTERGENESIZE     = 700;
my $INTERGENEFLANKORF    = 180;   # length in nts of intergene flanks borrowed 
                                  # from neighbor ORFs
</PRE>

<P>

<DIV ALIGN="CENTER"><A NAME="fig:flanks"></A><A NAME="290"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 6:</STRONG>
Two divergent ORFs flanking an intergenic region. Only 180 bases from each ORF are taken for compiling intergenic
clusters.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="820" HEIGHT="63" ALIGN="BOTTOM" BORDER="0"
 SRC="./flankingORFs.png"
 ALT="Image flankingORFs">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>
</LI>
<LI><code>-f</code> filters out cluster sequences with large differences in length. This flag 
compares sequences within a cluster to the first (arbitrary) reference sequence. Those with length difference 
(either shorter or longer) beyond the selected threshold will be removed. This might cause 
the resulting cluster to be entirely removed if the final number of taxa falls below the <code>-t</code> minimum.

<P>
</LI>
<LI><code>-r</code> allows the choice of any input genome (of course included in <code>-d</code> folder) 
as the reference, instead of the default smaller one. If possible, resulting clusters are named using gene names from 
this genome, which can be used to select well annotated species for this purpose. In addition, when using the default 
BDBH algorithm, the reference proteome is the one chosen to start adding genes in the clustering process. Therefore,
when using BDBH, the choice of reference proteome can have a large impact on the resulting number of clusters. By default,
the taxon with least genes is taken as reference. It is possible to change the way clusters are named by editing subroutine 
<code>extract_gene_name</code> in file <code>lib/phyTools.pm</code>.

<P>
</LI>
<LI><code>-e</code> excludes clusters with inparalogues, defined as sequences with best hits in its own genome. 
This option might be helpful to rule out clusters including several sequences from the same species, which might be of 
interest for users employing these clusters for primer design, for instance.

<P>
</LI>
<LI><code>-x</code> allows COG-generated sequence clusters to contain the same sequence in more than one cluster.

<P>
</LI>
<LI><code>-F</code> is the inflation value that governs Markov Clustering in OMCL runs, as explained in 
PubMed=http://www.ncbi.nlm.nih.gov/sites/ppmc/articles/PMC403725/12952885. As a rule of thumb,
low inflation values (<code>-F 1</code>)result in the inclusion of more sequences in fewer groups, whilst large values 
produce more, smaller clusters (<code>-F 4</code>).

<P>
</LI>
<LI><code>-A</code> tells the program to produce a tab-separated file with average % sequence identity values among pairs of genomes, 
computed from sequences in the final set of clusters (see also option <code>-t </code>). 
By default these identities are derived from BLASTP alignments, and hence correspond to amino acid sequence identities. 
However, as explained earlier, option -a forces the program to use nucleotide sequences and run BLASTN
instead, and therefore, -a 'CDS' combined -A will produce genomic average nucleotide sequence identities (ANI), as used
in the literature to help define prokaryotic species 
(PubMed=http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2776425/19855009). 

<P>
</LI>
<LI><code>-z</code> can be called when performing a genome composition analysis with clustering algorithms OMCL or COGS.
In addition to the core- and pan-genome tab-separated files mentioned earlier (see option <code>-c</code>), this flag requests 
a soft-core report, considering all sequence clusters present in a fraction of genomes defined by global variable <code>$SOFTCOREFRACTION</code>,
with a default value of 0.95. This choice produces a composition report more robust to assembly or annotation errors than the core-genome.

<P>
</LI>
</UL>

<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A NAME="tex2html225"
  HREF="node12.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html221"
  HREF="node7.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html215"
  HREF="node10.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html223"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html226"
  HREF="node12.html">Accompanying scripts</A>
<B> Up:</B> <A NAME="tex2html222"
  HREF="node7.html">User manual</A>
<B> Previous:</B> <A NAME="tex2html216"
  HREF="node10.html">(Eukaryotic) FASTA amino acid</A>
 &nbsp; <B>  <A NAME="tex2html224"
  HREF="node1.html">Contents</A></B> </DIV>
<!--End of Navigation Panel-->
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
