<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Run options</TITLE>
<META NAME="description" CONTENT="Run options">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node33.html">
<LINK REL="previous" HREF="node31.html">
<LINK REL="up" HREF="node30.html">
<LINK REL="next" HREF="node33.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html497"
  HREF="node33.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html493"
  HREF="node30.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html487"
  HREF="node31.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html495"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html498"
  HREF="node33.html">Downstream analyses</A>
<B> Up:</B> <A NAME="tex2html494"
  HREF="node30.html">Frequently asked questions (FAQs)</A>
<B> Previous:</B> <A NAME="tex2html488"
  HREF="node31.html">Installation</A>
 &nbsp; <B>  <A NAME="tex2html496"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H2><A NAME="SECTION00062000000000000000">
Run options</A>
</H2> 

<P>

<UL>
<LI>Why is it that I can't re-use BLAST results from the original <SPAN  CLASS="textit">get_homologues-est.pl</SPAN> after updating to <SPAN  CLASS="textit">get_homologues-est.pl</SPAN>?

<P>
In the v2 code family BLAST parsing evolved to capture -outfmt 6 'qseqid sseqid pident length qlen slen qstart qend sstart send evalue bitscore'. 
However, the original code base was capturing different data columns and this implies that old BLAST results from previous releases won't work anymore.

<P>
</LI>
<LI>Is there any real scenario in that the default value for -S flag is the best option?

<P>
Option -S 1 means that two sequences can be considered orthologues or inparalogues with any sequence identity, 
provided that <!-- MATH
 $Evalue < max(Evalue)$
 -->
<SPAN CLASS="MATH"><IMG
 WIDTH="171" HEIGHT="32" ALIGN="MIDDLE" BORDER="0"
 SRC="img3.png"
 ALT="$Evalue &lt; max(Evalue)$"></SPAN> and alignment coverage is sufficient (<code>-C</code>). This can be useful when
comparing very divergent genomes/proteomes, where orthologues can be distant in terms of sequence. 
If a reciprocal hit between genomes A and B is, say 30% identical, still it is a likely orthologue. 
It could of course be a false positive, and that's why you need to know, perhaps by using <code>-A</code>, the average identity
among proteins in those genomes, and ultimately a gene phylogenetic tree, including species A, B and others to compare
 to the species tree.

<P>
</LI>
<LI>Could you please explain a bit more what option <code>-o</code> is good for?

<P>
This option is useful in situations like the following: you have access to a compute cluster with modern compute nodes (64bit cores and a good amount of memory), 
but the master computer is an old, 32bit machine with limited RAM (say 1GB or less). Under such a setting, it would be convenient to distribute the first steps 
of the pipeline (all against all BLAST jobs) on the cluster using options <code>-o -m cluster</code>. The downstream parsing of the BLAST results, which is 
memory-consuming, could then be executed logged in from one of the more powerful compute nodes or from a powerful server.

<P>
</LI>
<LI>Would it be possible, after running the all-against-all BLAST jobs using option <code>-o</code>, to log into three different compute nodes 
and execute the downstream pipeline on each node for a different clustering algorithm (i.e. BDBHs, COG and OMCL)?

<P>
No, as each <SPAN  CLASS="textit">get_homologues.pl</SPAN> job requires exclusive access to the data directory and prevents other jobs to access it simultaneously.

<P>
</LI>
<LI>Does the -I option have any impact on the calculation of the core and pan genomes, including their graphical representations?

<P>
Of course, as this option enforces the program to sample once the list of genomes in the implicit order of the list. The fitting of pan-/core-genome
functions will be affected as there will be less points to do the noon-linear fitting.

<P>
</LI>
<LI>What is the minimum value for BLAST neighborhood correlation parameter?

<P>
This parameter captures the concept of BLAST neighborhood explained in 
PubMed=http://www.ncbi.nlm.nih.gov/sites/ppmc/articles/PMC2377100/18475320, and is calculated as a  
a positive Pearson correlation coefficient. In this context, sequences A and B should have a similar list 
of BLAST matches if they truly are orthologues.
This parameter is by default set to 0, so it takes no effect. As its values approaches to 1, it gets more difficult to call
bidirectional best hits, as they will be required to have very similar lists of BLAST hits.

<P>
</LI>
<LI>Could you explain a bit more about the meaning/effect of the <code>-t</code> option?

<P>
You should use <code>-t 0</code> (zero) when you need to get all clusters of homologous sequences. 
This will generate files containing 2 or more homologous sequences from one or more taxa. 
By default, <SPAN  CLASS="textit">get_homologues.pl</SPAN> sets t=numberOfTaxa, that is, it will provide the user with clusters of homologous sequences 
that contain at least one sequence from each taxon. Note that singleton clusters (just one sequence per organism) will be produced  
if combined with option <code>-e</code>, which excludes clusters containing inparalogues. In this later case, the resulting clusters will 
contain only single copy genes from each taxon, i.e. the orthologues. This is convenient if you want to use resulting gene families 
to do for example genome-level phylogenetic analyses using only the repertoire of orthologous single copy genes. Note, however, that  
such clusters are not suitable for pangenome analyses. For such analyses, instead, please use auxiliary script <SPAN  CLASS="textit">compare_clusters.pl</SPAN> 
with the set of clusters obtained with <code>-t 0</code>.

<P>
</LI>
<LI>And what about option -a, what are these other GenBank file features?

<P>
Thoroughly annotated genomes (see for example the GenBank file for Escherichia coli K12 MG1655: Escherichia_coli_K_12_substr__MG1655_uid225.gbk) 
have many features, incluing ribosomal rRNAs, integenic spacers, tRNAas, 'mat_peptide', 'repeat_region', and miscellaneous features, noted as 'misc_feature', 
in the corresponding GenBank files. Miscellaneous features report on diverse things such as cryptic prophages, the target sites of resolvases involved 
in replicon replication,  and many more "miscellaneous" features. Note that there is great heterogeneity both in the format and detail of annotating 
genome features. By default, <SPAN  CLASS="textit">get_homologues.pl</SPAN> will extract only the 'CDS' feature (i.e. the sequences for protein-coding sequences) from the 
GenBank file. By using option <code>-a</code> the user can select which features to parse, such as 'tRNA' or 'rRNA'. 

<P>
</LI>
<LI>Could you explain what the inflation parameter of orthoMCL is and how to decide what value to use?

<P>
The original OrthoMCL paper (PubMed=http://www.ncbi.nlm.nih.gov/sites/ppmc/articles/PMC403725/12952885) explains it:
"[...] changing the inflation index affects cluster tightness: Lower inflation values result in the inclusion of more sequences in fewer groups, 
whereas increasing the inflation index fragments clusters and reduces the number of sequences included". 
In our benchmarks with bacterial genomes this parameter shows a very modest, negligible effect.  

<P>
</LI>
<LI>I've run the <SPAN  CLASS="textit">get_homologues.pl</SPAN> pipeline some time ago. I would like to add some new genomes to the previous analysis. 
How do I proceed so as to reuse as much of the previous computations as possible?

<P>
If you still conserve the original input folder with FASTA or GenBank sequence files and the results <code>_homologues</code> directory,
both contained in the same directory, all you will have to do is to copy the new sequence files to the input folder and re-run
<SPAN  CLASS="textit">get_homologues.pl</SPAN>. This will ensure that only new required BLAST/PFam searches are completed, conserving the previous results as much 
as possible.

<P>
</LI>
<LI>How does <SPAN  CLASS="textit">get_homologues.pl</SPAN> decide how to name a certain cluster file? Is this affected by the use of -r?

<P>
Cluster files are named using gene names from the reference genome, or from the first included genome otherwise.
If a given genome R is selected with option <code>-r</code> then gene names from R will be used preferably. 
It is possible to change the way clusters are named by editing subroutine <code>extract_gene_name</code> in file <code>lib/phyTools.pm</code>.

<P>
</LI>
<LI>What happens if I perform the above explained steps but using a different reference genome?

<P>
The most obvious effect is that any resulting clusters will now be named according to gene names of the new reference genome.
A more subtle consequence for BDBH jobs is that now all genomes are compared to this reference, see figure <A HREF="node11.html#BDBHflow">3</A>,
and this will change the order in which bidirectional best hits are computed.

<P>
</LI>
<LI>How are the gene clusters named if no -r reference genome is specified?

<P>
In this case the genome with the least number of genes/features will be taken as the reference,
and resulting sequence clusters will be named according to gene names of this reference genome, 
which might not be the best annotated genome in your set. For this reason it is often a good idea to set as reference genome
one with a good annotation, for instance the species or strain described in http://www.ncbi.nlm.nih.gov/refseq/RefSeq.

<P>
</LI>
<LI>I have 40 draft genomes annotated in gbk format and I am using <SPAN  CLASS="textit">get_homologues</SPAN> to obtain the core and pan genomes. 
My plan is to run <SPAN  CLASS="textit">get_homologues</SPAN> with BDBH and -b to speed up the process. How can I choose the most appropriate reference genome?

<P>
Option <code>-b</code> is only suited for core-genome calculation, not pangenome. If this is really the desired task, the genome with the least number 
of annotated genes should be used as a reference, which is what the program would do by default, or else the best-annotated among small genomes.
However, note that this sort of core-genome calculation is most sensitive to missing genes, usually due to poor automatic annotations, which is 
why compiling a pangenome matrix is recommended when possible (see Section <A HREF="node22.html#pange">4.8.1</A>), so that a more robust soft-core can be estimated.

<P>
</LI>
<LI>Why does option '-t 0' not work with BDBH in <SPAN  CLASS="textit">get_homologues.pl</SPAN>? Is this also the reason BDBH cannot be used in a pangenome matrix analysis?

<P>
The reason is that BDBH uses a single reference genome and thus by definition cannot track genes not present in the reference. 
Therefore, pangenome matrices produced by the BDBH algorithm would be incomplete, considering only clusters including genes from the reference genome.
For this reason BDBH is adequate for core-genome calculation, but not for pan-genomes. 

<P>
</LI>
<LI>When the initial BLAST is being performed, does <SPAN  CLASS="textit">get_homologues.pl</SPAN> take into account the database size 
(i.e. the number of genes being BLASTed), or because the BLASTing is not done as an all-vs-all manner, do you not consider this as a factor in the analysis?

<P>
Within <code>lib/marfil_homology.pm</code> there is a global variable <code>$BLAST_DB_SIZE</code> set to 100_000_000 for that purpose.
That value is the fixed effective search space during BLAST searches so that any resulting E-values are comparable, even across experiments
or algorithm (BDBH, OMCL, COGS).

<P>
</LI>
<LI>I would like to get information of inparalogues and orthologues from each genome. I found several files in the tmp directory, 
such as <code>inparalogues_Buch_aph_APS.faa</code>. Could you explain about the files or a method to extract the information?

<P>
According to our working definition, all sequences grouped together in the same .faa/.fna cluster are likely orthologues, 
although you should always keep in mind that orthology is an evolutionary concept and therefore sequence-based approaches 
such as those in <SPAN  CLASS="textit">get_homologues.pl</SPAN> are simpler approximations. What is different with inparalogues? 
They are supposed to be duplicated genes that appeared after species separation, and therefore their orthology relationships 
are many-to-one or many-to-many. Inparalogues are easy to spot in clusters produced by <SPAN  CLASS="textit">get_homologues.pl</SPAN> because 
they are 2+ sequences from the same genome in the same cluster cluster. If you wish to know which inparalogue is most similar 
to an orthologous gene the best option is to run <SPAN  CLASS="textit">check_BDBHs.pl</SPAN>, which is explained on Section <A HREF="node28.html#checkBDBH">4.8.7</A>.

<P>
</LI>
<LI>I have been using GET_HOMOLOGUES and I could not figure out which is the default value for saving blastp hits, I mean, 
the value set for '-max_target_seqs'

<P>
BLAST parameter <code>-max_target_seqs</code> is set to the number of sequences of the query proteome, which usually is a large number 
that ensures all good quality hits are recovered.

<P>
</LI>
<LI>How does buffer flushing affects <SPAN  CLASS="textit">get_homologues.pl</SPAN>?

<P>
Although <SPAN  CLASS="textit">get_homologues.pl</SPAN> scripts explicitily flush output buffers (set with <code>$|=1</code>), 
users can occasionally experience buffering problems when writing to slow, external hard drives, as output files are often very large. 
Such problems have been reported when calling <code>-G</code> option, which in turn invokes subroutine <code>find_COGs</code> and calls several
external binaries, whose buffers cannot be flushed from the scripts. If that happens to you please consider increasing the sleep time
in that sub, which by default is 10 seconds.

<P>
</LI>
</UL>

<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A NAME="tex2html497"
  HREF="node33.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html493"
  HREF="node30.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html487"
  HREF="node31.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html495"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html498"
  HREF="node33.html">Downstream analyses</A>
<B> Up:</B> <A NAME="tex2html494"
  HREF="node30.html">Frequently asked questions (FAQs)</A>
<B> Previous:</B> <A NAME="tex2html488"
  HREF="node31.html">Installation</A>
 &nbsp; <B>  <A NAME="tex2html496"
  HREF="node1.html">Contents</A></B> </DIV>
<!--End of Navigation Panel-->
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
