<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Estimating average identity matrices</TITLE>
<META NAME="description" CONTENT="Estimating average identity matrices">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node28.html">
<LINK REL="previous" HREF="node26.html">
<LINK REL="up" HREF="node21.html">
<LINK REL="next" HREF="node28.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html438"
  HREF="node28.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html434"
  HREF="node21.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html428"
  HREF="node26.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html436"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html439"
  HREF="node28.html">Finding out best hits</A>
<B> Up:</B> <A NAME="tex2html435"
  HREF="node21.html">Performing genome composition analyses</A>
<B> Previous:</B> <A NAME="tex2html429"
  HREF="node26.html">Calculating Pfam enrichment of</A>
 &nbsp; <B>  <A NAME="tex2html437"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H3><A NAME="SECTION00058600000000000000"></A><A NAME="avgidmatrix"></A>
<BR>
Estimating average identity matrices
</H3>

<P>
If we recall for a moment the example GenBank files analyzed on section <A HREF="node16.html#gbkfiles">4.3</A> we can
demonstrate how to calculate average identity matrices, which can then be used to compare genome members of
a pangenome. To do so we will add a few flags to the previous command, in addition to flag <code>-A</code>, which specifically
asks for an identity matrix to be calculated: <code>$ ./get_homologues.pl -d sample_plasmids_gbk -A -t 0 -M</code>

<P>
This will produce the following output:
<PRE>
[...]
# number_of_clusters = 193
# cluster_list = [...]/UnculturedbacteriumplasmidpRSB203_f0_0taxa_algOMCL_e0_.cluster_list
# cluster_directory = [...]/UnculturedbacteriumplasmidpRSB203_f0_0taxa_algOMCL_e0_

# average_identity_matrix_file = 
# [...]/UnculturedbacteriumplasmidpRSB203_f0_0taxa_algOMCL_e0_Avg_identity.tab
# NOTE: matrix computed on blastp results
</PRE>

<P>
Note that on this example the produced identity matrix was calculated with the BLASTP scores among protein sequences included on the resulting 
clusters (193). If average nucleotide identities are desired the command must be modified to:

<P>
<code>$ ./get_homologues.pl -d sample_plasmids_gbk -a 'CDS' -A -t 0 -M</code>

<P>
Such matrices can then be used to calculate heatmaps and dendrograms that capture how similar the coding sequences are among genomes.
An example of this would be:

<P>
<PRE>
cd sample_plasmids_gbk_homologues
../hcluster_matrix.sh -i EcoliST131plasmidpKC394_f0_0taxa_CDS_algOMCL_e0_Avg_identity.tab
</PRE>

<P>

<DIV ALIGN="CENTER"><A NAME="fig:ANImap"></A><A NAME="595"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 16:</STRONG>
Example heatmap derived from an average nucleotide identity matrix calculated with get_homologues.pl.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="1350" HEIGHT="900" ALIGN="BOTTOM" BORDER="0"
 SRC="./Avg_identity_heatmap.png"
 ALT="Image Avg_identity_heatmap">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A NAME="tex2html438"
  HREF="node28.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html434"
  HREF="node21.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html428"
  HREF="node26.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html436"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html439"
  HREF="node28.html">Finding out best hits</A>
<B> Up:</B> <A NAME="tex2html435"
  HREF="node21.html">Performing genome composition analyses</A>
<B> Previous:</B> <A NAME="tex2html429"
  HREF="node26.html">Calculating Pfam enrichment of</A>
 &nbsp; <B>  <A NAME="tex2html437"
  HREF="node1.html">Contents</A></B> </DIV>
<!--End of Navigation Panel-->
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
