<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<!--Converted with LaTeX2HTML 2008 (1.71)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Obtaining (bacterial) GenBank input files</TITLE>
<META NAME="description" CONTENT="Obtaining (bacterial) GenBank input files">
<META NAME="keywords" CONTENT="manual">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META NAME="Generator" CONTENT="LaTeX2HTML v2008">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="manual.css">

<LINK REL="next" HREF="node10.html">
<LINK REL="previous" HREF="node8.html">
<LINK REL="up" HREF="node7.html">
<LINK REL="next" HREF="node10.html">
</HEAD>

<BODY >

<DIV CLASS="navigation"><!--Navigation Panel-->
<A NAME="tex2html201"
  HREF="node10.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html197"
  HREF="node7.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html191"
  HREF="node8.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html199"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html202"
  HREF="node10.html">(Eukaryotic) FASTA amino acid</A>
<B> Up:</B> <A NAME="tex2html198"
  HREF="node7.html">User manual</A>
<B> Previous:</B> <A NAME="tex2html192"
  HREF="node8.html">Input data</A>
 &nbsp; <B>  <A NAME="tex2html200"
  HREF="node1.html">Contents</A></B> 
<BR>
<BR></DIV>
<!--End of Navigation Panel-->

<H2><A NAME="SECTION00042000000000000000"></A> <A NAME="downGenBank"></A>
<BR>
Obtaining (bacterial) GenBank input files
</H2>

<P>
The http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.htmlGenBank format 
is routinely used to describe genomic sequences, usually taking one file per chromosome or genomic 
contig. Each file contains a reference DNA genomic sequence plus a collection of genes and their products, 
making it possible to extract simultaneously the sequence of every ORF and its corresponding protein products. 

<P>
GenBank files are the recommended input format for bacterial sequences, as they permit the compilation
of DNA and protein sequences clusters, which might have different applications.

<P>
There are many ways to obtain GenBank files, starting by manual browsing and downloading from the
http://www.ncbi.nlm.nih.gov/genbankNCBI site, keeping in mind that <SPAN ID="txt167">full files</SPAN>, 
which include the reference nucleotide sequences, should be downloaded. In fact, <SPAN  CLASS="textit">get_homologues.pl</SPAN>
will fail to parse any ORF in summary GenBank files. 

<P>

<DIV ALIGN="CENTER"><A NAME="fig:genbankfull"></A><A NAME="172"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 1:</STRONG>
NCBI download widget showing the choice of 'GenBank (full)' format, 
which contains the raw reference nucleotide sequences.</CAPTION>
<TR><TD>
<DIV ALIGN="CENTER">
<IMG
  WIDTH="1187" HEIGHT="429" ALIGN="BOTTOM" BORDER="0"
 SRC="./genbankfull.png"
 ALT="Image genbankfull">

</DIV></TD></TR>
</TABLE>
</DIV>

<P>
Often users take <SPAN ID="txt176">custom-made GenBank files</SPAN>, resulting from in-house genome assemblies, to be analysed. 
In most cases genes from such files don't have GenBank identifiers assigned yet, and so we recommend 
adding the field <code>locus_tag</code> to each CDS feature so that parsed sequences can be properly identified.

<P>
For their use with <SPAN  CLASS="textit">get_homologues</SPAN>,
GenBank files for the same species (for example, from the main chromosome and from a couple of plasmids)
must be concatenated. For instance, the genomic sequences of <SPAN  CLASS="textit">Rhizobium etli CFN 42</SPAN> comprise
http://www.ncbi.nlm.nih.gov/genome?Db=genome&amp;term=rhizobiumwhich can be concatenated into a single <SPAN  CLASS="textit">_Rhizobium_etli_CFN42.gbk</SPAN> file.

<P>
In order to assist in this task this software package includes the accompanying script
<SPAN  CLASS="textit">download_genomes_ncbi.pl</SPAN>. We will explain its use by fetching some of the <SPAN  CLASS="textit">Yersinia pestis</SPAN>
genomic sequences used in a http://www.ncbi.nlm.nih.gov/pubmed/210375712010 paper by Morelli <SPAN  CLASS="textit">et al</SPAN>:

<P>
<PRE>
Group   Name         Accession Number    Status
0.PE2   Pestoides F  NC_009381           Completed Sanger genome			
0.PE3   Angola       NC_010159           Completed Sanger genome			
0.PE4   91001        NC_005810           Completed Sanger genome			
0.ANT2  B42003004    NZ_AAYU00000000     Draft Sanger genome 				
1.ANT1  UG05-0454    NZ_AAYR00000000     Draft Sanger genome (12.3X coverage)
1.ANT1  Antiqua      NC_008150           Completed Sanger genome			
1.IN3   E1979001     NZ_AAYV00000000     Draft Sanger genome 				
1.ORI1  CA88-4125    NZ_ABCD00000000     Draft Sanger genome 				
1.ORI1  FV-1         NZ_AAUB00000000     Draft Sanger genome				 
1.ORI1  CO92         NC_003143           Completed Sanger genome  		 
1.ORI2  F1991016     NZ_ABAT00000000     Draft Sanger genome				 
1.ORI3  IP674        ERA000177           Draft 454 genome (82X coverage)   
1.ORI3  IP275        NZ_AAOS00000000     Draft Sanger genome (7.6X coverage) 
1.ORI3  MG05-1020    NZ_AAYS00000000     Draft Sanger genome (12.1X coverage)
2.ANT1  Nepal516     NZ_ACNQ00000000     Draft Sanger genome				 
2.MED1  KIM          NC_004088           Completed Sanger genome  		 
2.MED2  K1973002     NZ_AAYT00000000     Draft Sanger genome
</PRE>

<P>
In order to use <SPAN  CLASS="textit">download_genomes_ncbi.pl</SPAN> is is necessary to feed it a text file listing which 
genomes are to be downloaded. The next examples show the exact format required, 
as does the bundled file <code>sample_genome_list.txt</code>.

<BR>
<BR>
First, it can be seen that completed genomes have NC accession numbers, and 
can be added to the list as follows:
<PRE>
NC_010159                   Yersinia_pestis_Angola
</PRE>
Other annotated genomes can be added using their assembly code, as in this example:
can be added to the list as follows:
<PRE>
GCA_000016445.1_ASM1644v1   Yersinia_pestis_Pestoides_F
</PRE>
Finally, draft WGS genomes, which can be browsed at
<BR>
http://www.ncbi.nlm.nih.gov/Traces/wgshttp://www.ncbi.nlm.nih.gov/Traces/wgs,
can be listed in our download file by adding their four-letter code of their prefixes, as follows:
<PRE>
AAYU01                      Yersinia_pestis_B42003004
</PRE>

<P>
Finally, the <code>genome_list.txt</code> file will look as this:
<PRE>
NC_010159                  Yersinia_pestis_Angola    
GCA_000016445.1_ASM1644v1  Yersinia_pestis_Pestoides_F
AAYU01                     Yersinia_pestis_B42003004
</PRE>
Note that only the first two columns (separated by blanks) are read in, and that lines can be commented out
by adding a '#' as the first character. 
<BR>
<P>
Now we can run the following terminal command to fetch these genomes:
<BR><code>$ ./download_genomes_ncbi.pl genome_list.txt</code>
<BR>
which will put several <SPAN  CLASS="textit">_Yersinia_pestis_*.gbk</SPAN> files in the current directory, which are now
ready to be used by <SPAN  CLASS="textit">get_homologues</SPAN>.

<P>

<DIV CLASS="navigation"><HR>
<!--Navigation Panel-->
<A NAME="tex2html201"
  HREF="node10.html">
<IMG WIDTH="37" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="next"
 SRC="/usr/share/latex2html/icons/next.png"></A> 
<A NAME="tex2html197"
  HREF="node7.html">
<IMG WIDTH="26" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="up"
 SRC="/usr/share/latex2html/icons/up.png"></A> 
<A NAME="tex2html191"
  HREF="node8.html">
<IMG WIDTH="63" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="previous"
 SRC="/usr/share/latex2html/icons/prev.png"></A> 
<A NAME="tex2html199"
  HREF="node1.html">
<IMG WIDTH="65" HEIGHT="24" ALIGN="BOTTOM" BORDER="0" ALT="contents"
 SRC="/usr/share/latex2html/icons/contents.png"></A>  
<BR>
<B> Next:</B> <A NAME="tex2html202"
  HREF="node10.html">(Eukaryotic) FASTA amino acid</A>
<B> Up:</B> <A NAME="tex2html198"
  HREF="node7.html">User manual</A>
<B> Previous:</B> <A NAME="tex2html192"
  HREF="node8.html">Input data</A>
 &nbsp; <B>  <A NAME="tex2html200"
  HREF="node1.html">Contents</A></B> </DIV>
<!--End of Navigation Panel-->
<ADDRESS>
Bruno Contreras Moreira
2016-04-28
</ADDRESS>
</BODY>
</HTML>
